bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -612,12 +627,33 @@ public void run() {
                         Stopwatch stopwatch = Stopwatch.createStarted();
                         LOG.info("Starting PlacementPolicyCheck");
                         placementPolicyCheck();
+                        int numOfLedgersFoundInPlacementPolicyCheckValue = numOfLedgersFoundInPlacementPolicyCheck
+                                .get();
                         long placementPolicyCheckDuration = stopwatch.stop().elapsed(TimeUnit.MILLISECONDS);
-                        LOG.info("Completed placementPolicyCheck in {} milliSeconds", placementPolicyCheckDuration);
+                        LOG.info(
+                                "Completed placementPolicyCheck in {} milliSeconds."
+                                + " numOfLedgersNotAdheringToPlacementPolicy {}",
+                                placementPolicyCheckDuration, numOfLedgersFoundInPlacementPolicyCheckValue);
+                        ledgersNotAdheringToPlacementPolicyGuageValue
+                                .set(numOfLedgersFoundInPlacementPolicyCheckValue);
                         placementPolicyCheckTime.registerSuccessfulEvent(placementPolicyCheckDuration,
                                 TimeUnit.MILLISECONDS);
                     } catch (BKAuditException e) {
-                        LOG.error("BKAuditException running periodic placementPolicy check", e);
+                        int numOfLedgersFoundInPlacementPolicyCheckValue = numOfLedgersFoundInPlacementPolicyCheck
+                                .get();
+                        if (numOfLedgersFoundInPlacementPolicyCheckValue > 0) {
+                            /*
+                             * Though there is BKAuditException while doing
+                             * placementPolicyCheck, it found few ledgers not
+                             * adhering to placement policy. So reporting it.
+                             */
+                            ledgersNotAdheringToPlacementPolicyGuageValue
+                                    .set(numOfLedgersFoundInPlacementPolicyCheckValue);
+                        }
+                        LOG.error(
+                                "BKAuditException running periodic placementPolicy check."
+                                        + "numOfLedgersNotAdheringToPlacementPolicy {}",
+                                numOfLedgersFoundInPlacementPolicyCheckValue, e);
```

Comment:
@reddycharan Why are you checking for non zero value of numOfLedgersFoundInPlacementPolicyCheckValue to set the stat/guage but not for error message? Do you expect to have an exception but a zero numOfLedgersFoundInPlacementPolicyCheckValue?

Reply:
_Do you expect to have an exception but a zero numOfLedgersFoundInPlacementPolicyCheckValue?_

yes in the case of exception if we have not found faulty ledgers so far, we don't want to update the gauge value with zero. it would be misleading to report 0. so to avoid false negative i added this if condition in exception case.

Reply:
I see; what kind of exception do you expect even when numOfLedgersFoundInPlacementPolicyCheckValue is zero.

Reply:
for placementPolicyCheck, it has to iterate over all the ledgers in the metadataserver and check the placement policy. So half way though the process there could be an exception and no faulty ledger is found so far. So in this case we want to log error but dont want to report gauge with zero value.